x-pack/plugins/observability_solution/investigate_app/server/services/get_events.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import datemath from '@elastic/datemath';
+import { estypes } from '@elastic/elasticsearch';
+import { ElasticsearchClient } from '@kbn/core/server';
+import {
+  GetEventsParams,
+  GetEventsResponse,
+  getEventsResponseSchema,
+} from '@kbn/investigation-shared';
+import { Annotation } from '@kbn/observability-plugin/common/annotations';
+import { ScopedAnnotationsClient, unwrapEsResponse } from '@kbn/observability-plugin/server';
+import {
+  ALERT_REASON,
+  ALERT_RULE_CATEGORY,
+  ALERT_START,
+  ALERT_STATUS,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+import { AlertsClient } from './get_alerts_client';
+
+export function rangeQuery(
+  start: number,
+  end: number,
+  field = '@timestamp'
+): estypes.QueryDslQueryContainer[] {
+  return [
+    {
+      range: {
+        [field]: {
+          gte: start,
+          lte: end,
+          format: 'epoch_millis',
+        },
+      },
+    },
+  ];
+}
+
+export async function getAnnotationEvents(
+  params: GetEventsParams,
+  esClient: ElasticsearchClient,
+  annotationsClient: ScopedAnnotationsClient
+): Promise<GetEventsResponse> {
+  const startInMs = datemath.parse(params?.rangeFrom ?? 'now-15m')!.valueOf();
+  const endInMs = datemath.parse(params?.rangeTo ?? 'now')!.valueOf();
+  const filterJSON = params?.filter ? JSON.parse(params.filter) : {};
+
+  const body = {
+    size: 100,
+    query: {
+      bool: {
+        filter: [
+          ...rangeQuery(startInMs, endInMs),
+          ...Object.keys(filterJSON).map((filterKey) => ({
+            term: { [filterKey]: filterJSON[filterKey] },
+          })),
+        ],
+      },
+    },
+  };
+
+  try {
+    const response = await unwrapEsResponse(
+      esClient.search(
+        {
+          index: annotationsClient.index,
+          body,
+        },
+        { meta: true }
+      )
+    );
```

Comment:
i think it would make sense to use annotationClient find method here?

Reply:
The [find](https://github.com/elastic/kibana/blob/fd67f2d92ef299e11274c89a7195ed2cf6de1808/x-pack/plugins/observability_solution/observability/common/annotations.ts#L92) method currently doesn't accept `host.name` and in future we might also want to support other types of sources. Also, it fetches 10000 documents, I limited it to 100 for now. So it looks like we need to either update `find` API or keep our custom implementation. I think current implementation in PR is more flexible as it will support all types of sources without any modifications needed in future. Wdyt?

Reply:
I will recommend making the required changes on find method since we own it as well.

Reply:
I have updated the `find` method to support `filter` and `size`